### ISSUE TITLE
Pure attention model not training well

### ISSUE THREAD

**Issue by weiranwang19** [espnet/espnet#2690]
Hi,

This is not a bug but an intriguing observation (I accidentally applied the bug label and not sure how to remove that). I found the hybrid attention+ctc model work quite well in general, in both training and decoding. However, I have observed that the pure attention model does not work well with the default setup. Basically the attention model alone converges to pretty bad place in both loss and accuracy, and even the training progress is very slow. 

The attached figure shows the validation curves of alpha=0 (magenta curve, pure attention) and alpha=0.1 (gray curve, hybrid) for training on librispeech clean-100, and the units are characters.

![Screen Shot 2020-11-17 at 11 26 41 AM](https://user-images.githubusercontent.com/57633280/99438065-25579880-28c8-11eb-8930-e7b0153eeec9.png)

With just a little bit of CTC loss, the model trains fast and converges to much better places. But attention alone shall be a good model. I wonder if this is a known issue and what is the remedy. I changed the type of positional encoding and that does not change performance significantly.

Best,
Weiran

**Comment by sw005320** [comment]
Many thanks, @weiranwang19 for your report.
This is one of our TODO to provide a better setup for the attention-only mode.
Hybrid CTC-attention is quite stable in terms of the optimizer hyperparameters thanks to the CTC regularizers while attention could need a bit more tuning especially when the amount of training data is small (clean-100 corresponds to this case).

Is it based on transformer or RNN?
Anyway, if we change the optimizer hyperparameters (say tune the learning rate) or some other regularization hyperparameters, we could get some reasonable results.

**Comment by weiranwang19** [comment]
Hi @sw005320,

Thank you very much for your reply. Indeed I was able to make training going with a much smaller learning rate. See the attached plot (orange curve has alpha=0.3, and gray curve has alpha=0.0 pure attention):

![Screen Shot 2020-11-20 at 11 19 03 AM](https://user-images.githubusercontent.com/57633280/99840923-3ef41c80-2b22-11eb-9d03-23e3dc90687e.png)

Important parameters are as follows (for train_clean_100)
elayers: 14
dlayers: 6
adim: 512
aheads: 8
batch_count: seq
batch_size: 8
accum_grad: 4
ngpu: 8
transformer_lr: 1.0
transformer_warmup_steps: 10000
